pages/docs/reference/js-ir-compiler.md

```diff
@@ -0,0 +1,71 @@
+- Currently, the IR backend **does not generate source maps for Kotlin code**.
+- Some **libraries that rely on specific characteristics** of the default backend, such as `kotlin-wrappers`, can display some problems.
+- The IR backend **does not make Kotlin declarations available to JavaScript** by default at all. To make Kotlin declarations visible to JavaScript, they **must be** annotated with [`@JsExport`](js-to-kotlin-interop.html#jsexport-annotation).
+
+## Preview: Generation of TypeScript declaration files (d.ts)
+The Kotlin/JS IR compiler is capable of generating TypeScript definitions from your Kotlin code. These definitions can be used by JavaScript tools and IDEs when working on hybrid apps to provide autocompletion, support static analyzers, and make it easier to include Kotlin code in JavaScript and TypeScript projects.
+Top-level declarations marked with [`@JsExport`](js-to-kotlin-interop.html#jsexport-annotation) in a project that produces executable files (`binaries.executable()`) will get a `.d.ts` file generated, which contains the TypeScript definitions for the exported Kotlin code.
+In Kotlin 1.4, these declarations can be found in `build/js/packages/<package_name>/kotlin` alongside the corresponding, un-webpacked JavaScript code.
+
+The generation of TypeScript declaration files is a feature exclusive to the IR compiler, and is in active development. If you run into any problems, please submit them to the Kotlin [issue tracker](https://youtrack.jetbrains.com/issues?q=%23%7BKJS:%20d.ts%20generation%7D) or vote for submitted issues that impact you. 
+
+## Authoring libraries for the IR compiler with backwards compatibility
+
+If you're a library maintainer who is looking to provide compatibility with the default backend as well as the new IR compiler backend, a setting for the compiler selection is available that allows you to create artifacts for both backends, allowing you to keep compatibility for your exisiting users while providing support for the next generation of Kotlin compiler. This so-called `both`-mode can be turned on using the `kotlin.js.compiler=both` setting in your `gradle.properties` file, or can be set as one of the project-specific options inside your `js` block inside the `build.gradle(.kts)` file:
+
+```groovy
+kotlin {
+    js(BOTH) {
+        // . . .
+    }
+}
+```
+
+When in `both` mode, the IR compiler backend and default compiler backend are both used when building a library from your sources (hence the name). This means that both `klib` files with Kotlin IR as well as `js` files for the default compiler will be generated. When published under the same Maven coordinate, Gradle will automatically choose the right artifact depending on the use case – `js` for the old compiler, `klib` for the new one. This enables you to compile and publish your library for projects that are using either of the two compiler backends.
```

Comment:
I need to make a distinction here between `jar` and `klib`, because they're the quintessential two formats. I mainly mention this so that people can map their mental model to the file system – if it's a jar, it's for legacy, if it's a klib, it's for IR. Do you have suggestions what parts should be explained in more detail?

Reply:
For now, let's keep this distinction.